gobblin-runtime/src/main/java/gobblin/runtime/MysqlDatasetStateStore.java

```diff
@@ -64,12 +94,57 @@ public MysqlDatasetStateStore(DataSource dataSource, String stateStoreTableName,
    * @throws IOException if there's something wrong reading the {@link JobState.DatasetState}s
    */
   public Map<String, JobState.DatasetState> getLatestDatasetStatesByUrns(String jobName) throws IOException {
-    List<JobState.DatasetState> previousDatasetStates =
-        getAll(jobName, "%" + CURRENT_DATASET_STATE_FILE_SUFFIX + DATASET_STATE_STORE_TABLE_SUFFIX, true);
+    final String storeName = jobName;
```

Comment:
The issue I was facing was that the actual query needs to be different because we are trying to get the latest file for each Urn and the Urn is part of the name.  So, unless we do it in the query, we need to get all files, group by the Urn, and take the latest.